docs/ApiEndpoints.md

```diff
@@ -17,41 +17,41 @@
-| POST   | `/question/count-by-difficulty` | For given array of question IDs, <br> count how many questions in each difficulty   | `questionsId` (array)          | JWT token       | User            |
-| POST   | `/queue/join`                | Join the matching queue                         | `queueName` <br> `sessionID`                                          | JWT token       | User            |
-| POST   | `/queue/exit`                | Exit the matching queue                         | `queueName` <br> `sessionID`                                          | JWT token       | User            |
-| GET    | `/user/`                     | Get user information                            | `id` or `email`                                                       | JWT token       | User            |
-| DELETE | `/user/`                     | Delete user                                     | `id`                                                                  | JWT token       | User            |
-| GET    | `/user/all`                  | Get information for all users                   | -                                                                     | JWT token       | Maintainer      |
-| POST   | `/user/login`                | Login                                           | `email` <br> `password`                                               | -               | Guest           |
-| POST   | `/user/signup`               | create new user                                 | `email` <br> `password` <br> `confirmPassword`                        | -               | Guest           | 
-| PUT    | `/user/toggle-user-role`     | Toggle the user's role (user or maintainer)     | `id`                                                                  | JWT token       | Maintainer      |
-| PUT    | `/user/display-name`         | Update user's display name                      | `id` <br> `username`                                                  | JWT token       | User            |
-| PUT    | `/user/change-password`      | Change user's password                          | `id` <br> `currentPassword` <br> `newPassword` <br> `confirmPassword` | JWT token       | User            |
-| PUT    | `/user/programming-language` | Update user's preferred programming language    | `id` <br> `programmingLanguage`                                       | JWT token       | User            |
-| PUT    | `/user/complexity`           | Update user's preferred question complexity     | `id` <br> `complexity`                                                | JWT token       | User            |
+| Method | API Endpoints                     | Purpose                                                                           | Parameters <br> (JSON format)                                         | Header Contains | [Roles](#roles) |
```

Comment:
Did you try to 'format document' this page?

Reply:
![image](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g32/assets/63772723/46660861-82cb-4753-846f-5e2b83815388)

Already formatted on my page, might be a github display thing